lib/connection.js

```diff
@@ -377,6 +397,8 @@ class Connection extends EventEmitter {
       this, application, authStrategy, credentials,
       this._onSessionCreated.bind(this)
     );
+
+    this.emit('handshakeRequest', applicationName, authStrategy, credentials);
```

Comment:
@belochub what if we emit them only in debug mode, but in release mode we omit them?